### PR TITLE
Add tcp ping sender

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ environment:
 install:
   # install libsodium
   - mkdir deps && cd deps
-  - appveyor-retry appveyor DownloadFile https://download.libsodium.org/libsodium/releases/libsodium-1.0.13-mingw.tar.gz -FileName libsodium.tar.gz
+  - appveyor-retry appveyor DownloadFile https://download.libsodium.org/libsodium/releases/libsodium-1.0.16-mingw.tar.gz -FileName libsodium.tar.gz
```
Comment (kpp) [inline .appveyor.yml]: Split into separate commit to fix appveyor build

File: src/main.rs [hunk 2]
```diff
@@ -168,6 +170,21 @@ fn run_tcp(cli_config: &CliConfig, dht_sk: SecretKey, tcp_onion: TcpOnion) -> im
...
+    let interval = Duration::from_secs(1);
+    let wakeups = Interval::new(Instant::now(), interval);
+    let tcp_server_c = tcp_server.clone();
+
+    let tcp_ping_sender = wakeups
+        .map_err(|e| Error::new(ErrorKind::Other,e))
+        .for_each(move |_instant| {
+            trace!("Tcp server ping sender wake up");
+            tcp_server_c.send_pings()
+                .map_err(|e| e)
```
Comment (kurnevsky) [inline src/main.rs]: It seems it's no-op so can be removed.

File: src/main.rs [hunk 3]
```diff
@@ -168,6 +170,21 @@ fn run_tcp(cli_config: &CliConfig, dht_sk: SecretKey, tcp_onion: TcpOnion) -> im
         .map(|_| ())
         .map_err(|(e, _, _)| e);
 
+    let interval = Duration::from_secs(1);
```
Comment (kurnevsky) [inline src/main.rs]: Can we rename current `run` method inside `ServerExt` to `run_connection` and create new `run` that incorporates pings?

Comment (NamsooCho) [reply src/main.rs]: Good idea.

Comment (kpp) [reply src/main.rs]: `run` operates with only one connection so we will spam N pings for only 1 connection if we do that.

Comment (kurnevsky) [reply src/main.rs]: I meant to create a new `run` method that runs all connections.

Comment (NamsooCho) [reply src/main.rs]: Then we need to create a new Class. Right?
